### DISCUSSION TITLE
How to specify my own "model_state" input

### DISCUSSION THREAD

**Discussion by abhiroop513** [netket/netket#1354] [General]
Hello, 
How should I specify my own initial 'model_state' , only the initial inputs (the parameters will be generated randomly, as usual) for RBM or any Neural Network before any optimization calculation ? (The input may be other than binary)

**Comment by PhilipVinc** [comment]
Can you look if [the documentation](https://netket.readthedocs.io/en/latest/docs/varstate.html) or [the reference](https://netket.readthedocs.io/en/latest/api/_generated/vqs/netket.vqs.MCState.html#netket.vqs.MCState) answers your question, and if not can you rephrase what you are trying to do in a more clear way?

**Reply by abhiroop513** [reply]
For an RBM calculation (or, with any neural network), if I wish to specify my own input while defining the MC state, how should I do ?
`vs = nk.vqs.MCState(sa, model, n_samples=1000)
`

**Reply by abhiroop513** [reply]
Thank you for 'the reference'. I looked at it , as you mentioned. It seems the 'init_fun' initializes both the inputs and parameters for me. How should I do, if I want to give my own inputs, and init_fun will only initialize the parameters ?

**Reply by abhiroop513** [reply]
For instance, for a 10-site spin chain using RBM, if I want to have the following initial state as input:
[ 0.5, -0.5, -0.5, 0.5, 0.5, 0.5, -0.5, -0.5, -0.5, 0.5]

how should I give that ?

**Reply by PhilipVinc** [reply]
the question is what do you want to compute. The log-wave function, aka the model itself, or an expectation value with a certain set of samples?

**Reply by abhiroop513** [reply]
I wish to train the model of a neural network which will take the inputs other than binary (not the one-hot-encoded way), and I wish to specify the input myself. How can I do that ?
